### PR TITLE
AG-13464 Fix zoom toolbar hide when zoom module disabled

### PR DESCRIPTION
**Pull request by lsjroberts**
https://ag-grid.atlassian.net/browse/AG-13464

### CODE REVIEW

File: packages/ag-charts-enterprise/src/features/zoom/zoomModule.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ export const ZoomModule: _ModuleSupport.Module = {
     packageType: 'enterprise',
     chartTypes: ['cartesian', 'topology'],
     moduleFactory: (ctx) => new Zoom(ctx),
+    removable: false, // Toggling this module requires tear down of dom elements
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/features/zoom/zoomModule.ts]: This is an issue on all modules that have associated dom elements. Changing it to `false` fixes this, but does that leave another issue for which the modules were removed in the first place?

Comment (alantreadway) [reply packages/ag-charts-enterprise/src/features/zoom/zoomModule.ts]: This is probably a gap in the module implementations - they should handle removal correctly by destroying all resources they manage.

For the Sparklines implementation, we identified that all modules were being initialised, and so made a change to actually lazy initialise, and therefore destroy modules on an update to minimise additional resource allocations.

I'd prefer we fixed this by handling module destruction properly - and raise a distinct ticket for the other cases perhaps to be followed up seperately?

Comment (lsjroberts) [reply packages/ag-charts-enterprise/src/features/zoom/zoomModule.ts]: I've updated it to destroy the elements when the module is removed then recreate in the constructor, and hide them when it's just toggling the `zoom.buttons.enabled` property.

I'll use this pattern for the ranges and annotation toolbars.

**Review by alantreadway — APPROVED** [review]
LGTM - thanks!

**Pull request merged**